legacylogin/src/main/java/com/hamdan/forzenbook/legacy/login/LegacyLoginActivity.kt

```diff
@@ -1,35 +1,43 @@
 package com.hamdan.forzenbook.legacy.login
 
 import android.os.Bundle
+import android.view.inputmethod.EditorInfo
 import androidx.activity.ComponentActivity
 import androidx.activity.viewModels
 import androidx.core.view.isVisible
 import androidx.core.widget.addTextChangedListener
 import androidx.lifecycle.viewModelScope
 import com.hamdan.forzenbook.core.LoginError
+import com.hamdan.forzenbook.legacy.core.view.Navigator
 import com.hamdan.forzenbook.legacy.core.view.utils.DialogUtils.standardAlertDialog
+import com.hamdan.forzenbook.legacy.core.view.utils.KeyboardUtils
 import com.hamdan.forzenbook.legacy.core.viewmodels.LegacyLoginViewModel
 import com.hamdan.forzenbook.legacy.login.databinding.ActivityLegacyLoginBinding
 import com.hamdan.forzenbook.ui.core.R
 import dagger.hilt.android.AndroidEntryPoint
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
+import javax.inject.Inject
 
-// TODO add IME actions in FA-104
 @AndroidEntryPoint
 class LegacyLoginActivity : ComponentActivity() {
-    // TODO FA-104 add a click listener to send to Create Account on the createAccount text
+
+    @Inject
+    lateinit var navigator: Navigator
```

Comment:
You're giving the view control of its navigation logic. This needs to be in the ViewModel

Reply:
Got it, I wasn't sure if it belonged in the ViewModel or not because I wasn't sure if the navigation between pages should belong to the view.